packages/ui/src/components/Tooltip.tsx

```diff
@@ -0,0 +1,332 @@
+      };
+    case "inset-top-left":
+      return {
+        tooltipPosition: { top: 0, left: 0 },
+        trianglePosition: {},
+      };
+    case "inset-top-right":
+      return {
+        tooltipPosition: { top: 0, right: 0 },
+        trianglePosition: {},
+      };
+    case "inset-bottom-right":
+      return {
+        tooltipPosition: { bottom: 0, right: 0 },
+        trianglePosition: {},
+      };
+    default:
+      return { tooltipPosition: {}, trianglePosition: {} };
+  }
+};
+
+type UIPosition = {
+  top?: number;
+  bottom?: number;
+  left?: number;
+  right?: number;
+  width?: number;
+  height?: number;
+};
+
+type TooltipState = {
+  tooltipPosition?: UIPosition;
+  trianglePosition?: UIPosition;
+};
+
+type Position = {
+  x: number;
+  y: number;
+};
+
+type Size = {
+  width: number;
+  height: number;
+};
+
+export default (props: TooltipProps) => {
+  const { placement = "top", show, anchorRef, children } = props;
+
+  const tooltipRef = useRef<View | null>(null);
+  const containerRef = useRef<View | null>(null);
+
+  const [{ tooltipPosition, trianglePosition }, dispatch] = useReducer(
+    (
+      state: TooltipState,
+      { type, payload }: { type: string; payload: any },
+    ) => {
+      if (type === "init") {
+        return { ...getPositions(payload) };
+      }
+      if (type === "move") {
+        return {
+          tooltipPosition: { ...payload },
+        };
+      }
+      return state;
+    },
+    {},
+  );
+
+  const [containerPosition, setContainerPosition] = useState<Position | null>(
+    null,
+  );
+  useEffect(() => {
+    if (containerRef.current) {
+      containerRef.current.measure((x, y, width, height, pageX, pageY) => {
+        setContainerPosition({
+          x: pageX,
+          y: pageY,
+        });
+      });
+    }
+  }, [containerRef]);
+
+  const [anchorMeasure, setAnchorMeasure] = useState<Size | null>(null);
+  useEffect(() => {
+    if (anchorRef.current) {
+      anchorRef.current.measure((x, y, width, height) => {
+        setAnchorMeasure({
+          width,
+          height,
+        });
+      });
+    }
+  }, [anchorRef]);
+
+  useEffect(() => {
+    if (anchorMeasure?.width && tooltipRef.current) {
+      tooltipRef.current.measure((x, y, width, height) => {
+        dispatch({
+          type: "init",
+          payload: {
+            placement,
+            anchor: anchorMeasure,
+            tooltip: { width, height },
+          },
+        });
+      });
+    }
+  }, [anchorMeasure, show, placement]);
+
+  const trackCursor = useCallback(
+    (e: React.MouseEvent) => {
+      if (!containerPosition) return;
+      dispatch({
+        type: "move",
+        payload: {
+          top: e.clientY - containerPosition.y,
+          left: e.clientX - containerPosition.x,
+        },
+      });
+    },
+    [containerPosition],
+  );
+
+  const mouseEvents = useMemo(
+    () =>
+      placement === "cursor"
+        ? {
+            onMouseMove: trackCursor,
```

Comment:
Bad news here: `onMouseMove` isn't available in React Native! It happens to work in the Storybook because React Native Web implements it, but it won't work in the desktop app. Looks like we can use [react-native-hoverable](https://github.com/ryanlntn/react-native-hoverable). But before we go down that road, let me suggest: I think this is probably unnecessary! I'm fine—at least for now—with anchoring the tooltip to the container view based on its fixed position, rather than having the tooltip follow the mouse. Would you be okay with that simpler approach?

Reply:
Yeup. I added this last for no particular reason other than tinkering.